### PR TITLE
feat(object storage): add is_https_endpoint for minio

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/object_store/src/object/s3.rs [hunk 1]
```diff
@@ -617,8 +617,8 @@ impl S3ObjectStore {
         let server = server.strip_prefix("minio://").unwrap();
         let (access_key_id, rest) = server.split_once(':').unwrap();
         let (secret_access_key, rest) = rest.split_once('@').unwrap();
-        let (address, bucket) = rest.split_once('/').unwrap();
-
+        let (address, bucket) = rest.split_once('#').unwrap();
+        println!("111 {:?}", address);
```
Comment (hzxa21) [inline src/object_store/src/object/s3.rs:621]: remove

File: src/object_store/src/object/s3.rs [hunk 2]
```diff
@@ -617,8 +617,8 @@ impl S3ObjectStore {
         let server = server.strip_prefix("minio://").unwrap();
         let (access_key_id, rest) = server.split_once(':').unwrap();
         let (secret_access_key, rest) = rest.split_once('@').unwrap();
-        let (address, bucket) = rest.split_once('/').unwrap();
-
+        let (address, bucket) = rest.split_once('#').unwrap();
```
Comment (hzxa21) [inline src/object_store/src/object/s3.rs:620]: This change breaks compatibility meaning that deployment running prior to this PR will break. Can we make it compatible? For example, we can keep the original format for the url for http and use the following one for https:
```
hummock+minio_https://hummockadmin:hummockadmin@127.0.0.1:9301/hummock001
```

File: src/object_store/src/object/s3.rs [hunk 3]
```diff
@@ -629,7 +629,7 @@ impl S3ObjectStore {
 
         let config = builder
             .region(Region::new("custom"))
-            .endpoint_url(format!("http://{}", address))
+            .endpoint_url(address)
```
Comment (hzxa21) [inline src/object_store/src/object/s3.rs:632]: ditto

File: src/object_store/src/object/s3.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/object_store/src/object/s3.rs:620]: `hummock+minio_https://hummockadmin:hummockadmin@127.0.0.1:9301/hummock001` also break the compatibility as we will `server.strip_prefix("minio://").unwrap();`.

Comment (wcy-fdu) [reply src/object_store/src/object/s3.rs:620]: I think regex can be used to avoid compatibility issues, let me try it,

Comment (hzxa21) [reply src/object_store/src/object/s3.rs:620]: We can match `minio_https` first before `minio`. regex also works.

Comment (wcy-fdu) [reply src/object_store/src/object/s3.rs]: Yeap, no matter which modification method is used, corresponding changes are required on the operator side.

**Comment by zwang28** [comment]
Can we just simply remove the hardcode `http://`, and use whatever given:
1. If the given endpoint url is with scheme (http or https), use it as it it.
2. If the given endpoint url doesn't specify any scheme, append http scheme for it.

**Comment by wcy-fdu** [comment]
> Can we just remove the hardcode `http://`, and ensure backward compatibility via:
> 
> 1. If the given endpoint url is with scheme (http or https), use it as it is.
> 2. If the given endpoint url doesn't specify any scheme, add http scheme for it.
> 
> For example
> 
> 1. https example: minio://https://127.0.0.1:9301/hummock001
> 2. http example: minio://http://127.0.0.1:9301/hummock001 is the same as minio://127.0.0.1:9301/hummock001

Thanks for your suggestion!

**Review by zwang28 — APPROVED** [review]
LGTM. Maybe add some tests for the endpoint parser.

**Pull request merged**